src/backend/egl/display.rs

```diff
@@ -212,6 +201,108 @@ impl EGLDisplay {
         })
     }
 
+    /// Create a new [`EGLDisplay`] from an already initialized EGLDisplay and EGLConfig handle
+    ///
+    /// # Safety
+    ///
+    /// - Not using the system default EGL library (`dlopen("libEGL.so")`) as loaded by smithay might cause undefined behavior
+    /// - Display and config handles must be externally managed to ensure they do not become invalid before the compositor is shut down
```

Comment:
The handles created from the display (`get_display_handle`) need to be mentioned in the docs, the suggestion includes that

```suggestion
    /// - The `display` and `config` must be valid for the lifetime of the returned display and any handles created by this display (using [`EGLDisplay::get_display_handle`])
```